cpp/unit/ability.cpp

```diff
@@ -81,9 +88,23 @@ GarrisonAbility::GarrisonAbility(Sound *s)
 }
 
 bool GarrisonAbility::can_invoke(Unit &to_modify, const Command &cmd) {
-	return to_modify.location &&
-	       cmd.has_unit() &&
-	       cmd.unit()->has_attribute(attr_type::garrison);
```

Comment:
You could keep the indent depth lower by instead writing

    if (!cmd.has_unit()) {
        return false;
    }